test/File/ExcludeMimeTypeTest.php

```diff
@@ -25,17 +25,21 @@ public function basicBehaviorDataProvider()
     {
         $testFile = __DIR__ . '/_files/picture.jpg';
         $fileUpload = [
-            'tmp_name' => $testFile, 'name' => basename($testFile),
-            'size' => 200, 'error' => 0, 'type' => 'image/jpeg'
+            'tmp_name' => $testFile,
+            'name'     => basename($testFile),
+            'size'     => 200,
+            'error'    => 0,
+            'type'     => 'image/jpeg',
         ];
+
         return [
-            //    Options, isValid Param, Expected value
-            ['image/gif',                      $fileUpload, true],
-            ['image',                          $fileUpload, false],
-            ['test/notype',                    $fileUpload, true],
-            ['image/gif, image/jpeg',          $fileUpload, false],
+            //    Options, isValid Param, Expected value, messages
+            ['image/gif',                 $fileUpload, true],
+            ['image',                     $fileUpload, false, [ExcludeMimeType::FALSE_TYPE => "File has an incorrect mimetype of 'image/jpeg'"]],
+            ['test/notype',               $fileUpload, true],
+            ['image/gif, image/jpeg',     $fileUpload, false, [ExcludeMimeType::FALSE_TYPE => "File has an incorrect mimetype of 'image/jpeg'"]],
```

Comment:
https://github.com/zendframework/zend-inputfilter/blob/master/test/InputTest.php#L688

This is an example of a readable data provider. Notice even we mask the scalar values with a variable name so you can understand the meaning inmediatelly.